sys/include/ztimer/xtimer_compat.h

```diff
@@ -22,6 +22,10 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include "assert.h"
```

Comment:
Assert is a POSIX header (even though we redefine it in `core`)

```suggestion
#include <assert.h>
```